stock-service/src/main/java/rs/banka4/stock_service/service/impl/ActuaryServiceImpl.java

```diff
@@ -0,0 +1,224 @@
+            dto.limitAmount() != null
+                && dto.limitAmount()
+                    .compareTo(BigDecimal.ZERO)
+                    < 0
+        ) {
+            throw new NegativeLimitException(
+                dto.actuaryId()
+                    .toString()
+            );
+        }
+
+        ActuaryInfo actuaryInfo = new ActuaryInfo();
+        actuaryInfo.setUserId(dto.actuaryId());
+        actuaryInfo.setLimit(new MonetaryAmount(dto.limitAmount(), CurrencyCode.RSD));
+        actuaryInfo.setUsedLimit(new MonetaryAmount(BigDecimal.ZERO, CurrencyCode.RSD));
+        actuaryInfo.setNeedApproval(dto.needsApproval());
+        actuaryRepository.save(actuaryInfo);
+    }
+
+    @Override
+    public void changeActuaryDetails(UUID actuaryId, ActuaryPayloadDto dto) {
+
+        if (!actuaryId.equals(dto.actuaryId())) {
+            // case when we send the admin id as the path parameter also when all the securities
+            // should be transfered to the admin actuaryId is admins id
+        }
+
+
+        if (
+            dto.limitAmount() != null
+                && dto.limitAmount()
+                    .compareTo(BigDecimal.ZERO)
+                    < 0
+        ) {
+            throw new NegativeLimitException(actuaryId.toString());
+        }
+
+
+        ActuaryInfo actuaryInfo =
+            actuaryRepository.findById(dto.actuaryId())
+                .orElseThrow(
+                    () -> new ActuaryNotFoundException(
+                        dto.actuaryId()
+                            .toString()
+                    )
+                );
+
+        actuaryInfo.setNeedApproval(dto.needsApproval());
+        actuaryInfo.setLimit(new MonetaryAmount(dto.limitAmount(), CurrencyCode.RSD));
+        actuaryRepository.save(actuaryInfo);
+    }
+
+    @Override
+    public ResponseEntity<Page<CombinedResponse>> search(
+        Authentication auth,
+        String firstName,
+        String lastName,
+        String email,
+        String position,
+        int page,
+        int size
+    ) {
+
+        UserServiceClient userServiceClient = userServiceRetrofit.create(UserServiceClient.class);
+        String token = "Bearer " + auth.getCredentials();
+        try {
+            Response<PaginatedResponse<EmployeeResponseDto>> response =
+                userServiceClient.searchActuaryOnly(
+                    token,
+                    firstName,
+                    lastName,
+                    email,
+                    position,
+                    page,
+                    size
+                )
+                    .execute();
+
+            if (response.isSuccessful() && response.body() != null) {
+                PaginatedResponse<EmployeeResponseDto> employeePage = response.body();
+                if (
+
+                    employeePage.getContent()
+                        .isEmpty()
+                ) {
+                    logger.info("No actuaries found for the given search criteria.");
```

Comment:
this seems like a `.debug()` log (it'd also help to specify what the criteria were)